src/plugins/intel_cpu/src/nodes/rms_norm.cpp

```diff
@@ -0,0 +1,244 @@
+
+struct RMSNorm::RMSNormExecutor : public RMSNorm::Executor {
+    RMSNormExecutor(ov::element::Type precision, size_t data_size, size_t scale_size, float eps) : m_precision(precision) {
+        jit_rms_compile_params jcp;
+        jcp.src_prc = precision;
+        jcp.dst_prc = precision;
+        jcp.data_size = data_size;
+        jcp.scale_size = scale_size;
+        jcp.eps = eps;
+        m_kernel = createJitKernel(jcp);
+    }
+    void execute(const std::vector<MemoryPtr>& inputs, const MemoryPtr output) override {
+        auto src = inputs[0]->getDataAs<uint8_t>();
+        auto dst = output->getDataAs<uint8_t>();
+        float* scale = inputs[1]->getDataAs<float>();
+
+        const auto& src_strides = inputs[0]->getDescWithType<BlockedMemoryDesc>()->getStrides();
+        const auto& dst_strides = output->getDescWithType<BlockedMemoryDesc>()->getStrides();
+        const auto& shape = inputs[0]->getStaticDims();
+        const auto src_stride = src_strides[src_strides.size() - 2] * m_precision.size();
+        const auto dst_stride = dst_strides[dst_strides.size() - 2] * m_precision.size();
+        auto n = shape_size(shape) / shape[shape.size() - 1];
+        parallel_for(n, [&] (size_t i) {
+            execJitKernel(m_kernel, src + i * src_stride, dst + i * dst_stride, scale);
+        });
+    }
+
+private:
+    ov::element::Type m_precision;
+    std::shared_ptr<kernel::JitKernelBase> m_kernel;
+};
+
+RMSNorm::RMSNorm(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr context)
+    : Node(op, context, RMSNormShapeInferFactory(op)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        OPENVINO_THROW("CPU: " + errorMessage);
+    }
+    const auto rms = std::dynamic_pointer_cast<const ov::op::internal::RMS>(op);
+    m_eps = static_cast<float>(rms->get_epsilon());
+}
+
+void RMSNorm::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+    auto precision = getOriginalInputPrecisionAtPort(0);
+
+    impl_desc_type impl_type;
+    if (mayiuse(cpu::x64::avx512_core)) {
+        impl_type = impl_desc_type::jit_avx512;
+    } else if (mayiuse(cpu::x64::avx2)) {
+        impl_type = impl_desc_type::jit_avx2;
+    } else {
+        impl_type = impl_desc_type::ref;
+    }
+
+    addSupportedPrimDesc({{LayoutType::ncsp, precision}, {LayoutType::ncsp, ov::element::f32}},
+                        {{LayoutType::ncsp, precision}},
+                        impl_type);
+}
+
+void RMSNorm::createPrimitive() {
+    auto precision = getOriginalInputPrecisionAtPort(0);
+    auto data_dims = getSrcMemoryAtPort(0)->getDescWithType<BlockedMemoryDesc>()->getBlockDims();
+    size_t data_size = data_dims[data_dims.size() - 1];
+    size_t scale_size = shape_size(getSrcMemoryAtPort(1)->getDescWithType<BlockedMemoryDesc>()->getBlockDims());
+
+    RMSNormKey key = {precision, data_size, scale_size, static_cast<size_t>(dnnl::impl::float2int(m_eps))};
```

Comment:
why we need bit-cast `m_eps` from float to size_t here?

Reply:
Bit-cast removed due to float is also hash-able.